src/odemis/driver/test/simcam_test.py

```diff
@@ -302,6 +310,83 @@ def test_focus(self):
         f.result()
         self.assertEqual(self.focus.position.value, pos)
 
+
+class TestSimCamWithPolarization(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.camera = CLASS(**KWARGS_POL)
+        for child in cls.camera.children.value:
+            if child.name == CONFIG_FOCUS["name"]:
+                cls.focus = child
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.camera.terminate()
+
+    def setUp(self):
+        size = self.camera.shape[:-1]
+        self.is_rgb = (len(size) >= 3 and size[-1] in {3, 4})
+        # image shape is inverted order of size
+        if self.is_rgb:
+            self.imshp = size[-2::-1] + size[-1:] # RGB dim at the end
+        else:
+            self.imshp = size[::-1]
+        self.camera.resolution.value = self.camera.resolution.range[1]
+        self.acq_dates = (set(), set()) # 2 sets of dates, one for each receiver
+
+    def tearDown(self):
+        pass
+
+    def _ensureExp(self, exp):
+        """
+        Ensure the camera has picked up the new exposure time
+        """
+        old_exp = self.camera.exposureTime.value
+        self.camera.exposureTime.value = exp
+        time.sleep(old_exp) # wait for the last frame (worst case)
+
+    #     @unittest.skip("simple")
+    def test_acquire_ar_pol(self):
+        """
+        Acquire image with text of current polarization position written on top.
+        """
+        self.assertGreaterEqual(len(self.camera.shape), 3)
+        exposure = 0.1
+        self._ensureExp(exposure)
+
+        image = "../sparc-ar-mirror-align.h5"
+        # get image from camera
+        im = self.camera.data.get()
+        # create image with text for comparison
+        im_test = self._write_pol(image)
```

Comment:
Instead of comparing bit for bit that the text is correct, you could just move the pol-analyser to different positions, and check that the image generated is different from all the other positions.